liota/entities/devices/sensor_tag.py

```diff
@@ -0,0 +1,451 @@
+#  Licensed under the BSD 2-Clause License (the “License”); you may not use   #
+#  this file except in compliance with the License.                           #
+#                                                                             #
+#  The BSD 2-Clause License                                                   #
+#                                                                             #
+#  Redistribution and use in source and binary forms, with or without         #
+#  modification, are permitted provided that the following conditions are met:#
+#                                                                             #
+#  - Redistributions of source code must retain the above copyright notice,   #
+#      this list of conditions and the following disclaimer.                  #
+#                                                                             #
+#  - Redistributions in binary form must reproduce the above copyright        #
+#      notice, this list of conditions and the following disclaimer in the    #
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import time
+import Queue
+import logging
+from threading import Thread
+
+from aenum import UniqueEnum
+from bluepy.sensortag import SensorTag
+from bluepy.btle import BTLEException
+
+from liota.entities.devices.device import Device
+from liota.lib.utilities.utility import systemUUID
+
+log = logging.getLogger(__name__)
+
+
+class Sensors(UniqueEnum):
+    """
+    Enum for sensors that can be enabled in a SensorTag Device.
+
+    Note: Use ALL to enable all the sensors.
+    """
+    TEMPERATURE = 0
+    HUMIDITY = 1
+    BAROMETER = 2
+    ACCELEROMETER = 3
+    MAGNETOMETER = 4
+    GYROSCOPE = 5
+    LIGHTMETER = 6
+    BATTERY_LEVEL = 7
+    # All the above sensors
+    ALL = 8
+
+
+class SensorTagDevice(Device, SensorTag):
```

Comment:
Why is SensorTagDevice dependent on SensorTag ? Any specific requirement, I`m not clear with this implementation.